PWGHF/TableProducer/treeCreatorToXiPi.cxx

```diff
@@ -294,27 +344,167 @@ struct HfTreeCreatorToXiPi {
+
+      rowCandidateLite(
+        candidate.xPv(),
+        candidate.yPv(),
+        candidate.zPv(),
+        candidate.collision().numContrib(),
+        candidate.collision().chi2(),
+        candidate.xDecayVtxCharmBaryon(),
+        candidate.yDecayVtxCharmBaryon(),
+        candidate.zDecayVtxCharmBaryon(),
+        candidate.xDecayVtxCascade(),
+        candidate.yDecayVtxCascade(),
+        candidate.zDecayVtxCascade(),
+        candidate.xDecayVtxV0(),
+        candidate.yDecayVtxV0(),
+        candidate.zDecayVtxV0(),
+        candidate.signDecay(),
+        candidate.pxCharmBaryon(),
+        candidate.pyCharmBaryon(),
+        candidate.pzCharmBaryon(),
+        candidate.pxPiFromCharmBaryon(),
+        candidate.pyPiFromCharmBaryon(),
+        candidate.pzPiFromCharmBaryon(),
+        candidate.pxPiFromCasc(),
+        candidate.pyPiFromCasc(),
+        candidate.pzPiFromCasc(),
+        candidate.pxPosV0Dau(),
+        candidate.pyPosV0Dau(),
+        candidate.pzPosV0Dau(),
+        candidate.pxNegV0Dau(),
+        candidate.pyNegV0Dau(),
+        candidate.pzNegV0Dau(),
+        candidate.impactParCascXY(),
+        candidate.impactParPiFromCharmBaryonXY(),
+        candidate.errImpactParCascXY(),
+        candidate.errImpactParPiFromCharmBaryonXY(),
+        candidate.invMassLambda(),
+        candidate.invMassCascade(),
+        candidate.invMassCharmBaryon(),
+        candidate.etaV0PosDau(),
+        candidate.etaV0NegDau(),
+        candidate.etaPiFromCasc(),
+        candidate.etaPiFromCharmBaryon(),
+        candidate.dcaXYToPvV0Dau0(),
+        candidate.dcaXYToPvV0Dau1(),
+        candidate.dcaXYToPvCascDau(),
+        candidate.dcaCascDau(),
+        candidate.dcaV0Dau(),
+        candidate.dcaCharmBaryonDau(),
+        candidate.errorDecayLengthCharmBaryon(),
+        candidate.impactParCascXY() / candidate.errImpactParCascXY(),
+        candidate.impactParPiFromCharmBaryonXY() / candidate.errImpactParPiFromCharmBaryonXY(),
+        candidate.template piFromCharmBaryon_as<TMyTracks>().isGlobalTrackWoDCA(),
+        candidate.template piFromCharmBaryon_as<TMyTracks>().itsNCls(),
+        candidate.pidTpcInfoStored(),
+        candidate.pidTofInfoStored(),
+        candidate.tpcNSigmaPiFromCharmBaryon(),
+        candidate.tpcNSigmaPiFromCasc(),
+        candidate.tpcNSigmaPiFromLambda(),
+        candidate.tpcNSigmaPrFromLambda(),
+        candidate.tofNSigmaPiFromCharmBaryon(),
+        candidate.tofNSigmaPiFromCasc(),
+        candidate.tofNSigmaPiFromLambda(),
+        candidate.tofNSigmaPrFromLambda(),
+        flagMc,
+        originMc,
+        collisionMatched);
+    }
+  }
+
+  void processDataFull(MyEventTable const& collisions, MyTrackTable const&,
+                       soa::Join<aod::HfCandToXiPi, aod::HfSelToXiPi> const& candidates)
+  {
+    // Filling event properties
+    bool newEvent = true;
```

Comment:
When is `newEvent` set to `false`?

Reply:
This should never be set to false, it is filled in the table when a new collision is processed, so it will always be a new event --> this variable should be always true